meson.build

```diff
@@ -36,7 +36,11 @@ add_project_arguments(
 zlib = dependency('zlib')
 threads = dependency('threads')
 jansson = dependency('jansson', version: '>=2.10')
-libcrypto = dependency('libcrypto', version: '>=1.0.2')
+if host_machine.system() == 'freebsd'
+  libcrypto = meson.get_compiler('c').find_library('crypto', dirs: [ '/usr/local/lib/','/usr/lib/' ])
+else
+  libcrypto = dependency('libcrypto', version: '>=1.0.2')
```

Comment:
1.0.2 is EOL and 1.1.1 is approaching EOL in September. Is there a reason why this is still supporting outdated versions of OpenSSL?

Reply:
Probably an artifact from "olden days" (this patch has been pending for almost two years, not withstanding the recent change for openssl 3.)  I do remember having a build fail when a test tried to call something that wasn't in 1.0.2 and 1.0.2 was what was installed.  IIRC it was in one of the automated CI test builds. Given that 1.0.2 is now well past EOL, there's probably no reason to keep it.

Reply:
Note this patch really only moves that line into an if statement and didn't specifically add it.

Reply:
I did some testing and it seems to work in all cases with the default dependency check, so I removed all the complex logic.